File: src/main/java/painpoints/Pojo.java
package painpoints;

public class Pojo {
    private String user;
    private Integer age;

    public Pojo(String user) {
        this.user = user;
    }

    public Pojo(String user, Integer age) {
        this.user = user;
        this.age = age;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }
}

File: src/main/kotlin/solve/Pojo.kt
package solve

class Pojo(var user: String?, var age: Int? = null)

fun main(args: Array<String>) {
    var a = Pojo("a")
    val b = Pojo("b", 18)
    a = Pojo("A") // it works
//    b = Pojo("A") // compilation error
}
